x-pack/plugins/task_manager/server/lib/create_managed_configuration.ts

```diff
@@ -0,0 +1,162 @@
+    } else {
+      // Increase max workers by MAX_WORKERS_INCREASE_PERCENTAGE while making sure it doesn't go
+      // higher than the starting value. Using Math.ceil to make sure the number is different than
+      // previous while not being a decimal value
+      newMaxWorkers = Math.min(
+        startingMaxWorkers,
+        Math.ceil(previousMaxWorkers * MAX_WORKERS_INCREASE_PERCENTAGE)
+      );
+    }
+    if (newMaxWorkers !== previousMaxWorkers) {
+      logger.debug(
+        `Max workers configuration changing from ${previousMaxWorkers} to ${newMaxWorkers} after seeing ${errorCount} error(s)`
+      );
+      if (previousMaxWorkers === startingMaxWorkers) {
+        logger.warn(
+          `Max workers configuration is temporarily reduced after Elasticsearch returned ${errorCount} "too many request" error(s).`
+        );
+      }
+    }
+    return newMaxWorkers;
+  }, startingMaxWorkers);
+}
+
+function createPollIntervalScan(logger: Logger, startingPollInterval: number) {
+  return scan((previousPollInterval: number, errorCount: number) => {
+    let newPollInterval: number;
+    if (errorCount > 0) {
+      // Increase poll interval by POLL_INTERVAL_INCREASE_PERCENTAGE and use Math.ceil to
+      // make sure the number is different than previous while not being a decimal value.
+      newPollInterval = Math.ceil(previousPollInterval * POLL_INTERVAL_INCREASE_PERCENTAGE);
+    } else {
+      // Decrease poll interval by POLL_INTERVAL_DECREASE_PERCENTAGE and use Math.floor to
+      // make sure the number is different than previous while not being a decimal value.
+      newPollInterval = Math.max(
+        startingPollInterval,
+        Math.floor(previousPollInterval * POLL_INTERVAL_DECREASE_PERCENTAGE)
+      );
+    }
+    if (newPollInterval !== previousPollInterval) {
+      logger.debug(
+        `Poll interval configuration changing from ${previousPollInterval} to ${newPollInterval} after seeing ${errorCount} error(s)`
+      );
+      if (previousPollInterval === startingPollInterval) {
+        logger.warn(
+          `Poll interval configuration is temporarily increased after Elasticsearch returned ${errorCount} "too many request" error(s).`
+        );
+      }
+    }
+    return newPollInterval;
+  }, startingPollInterval);
+}
+
+function countErrors(errors$: Observable<Error>, countInterval: number): Observable<number> {
+  return merge(
+    // Flush error count at fixed interval
+    interval(countInterval).pipe(map(() => FLUSH_MARKER)),
+    errors$.pipe(filter((e) => SavedObjectsErrorHelpers.isTooManyRequestsError(e)))
+  ).pipe(
+    // When tag is "flush", reset the error counter
+    // Otherwise increment the error counter
+    mergeScan(
+      ({ count }, next) =>
+        next === FLUSH_MARKER
+          ? of(emitErrorCount(count), resetErrorCount())
+          : of(incementErrorCount(count)),
+      emitErrorCount(0)
```

Comment:
nit:  The placement of the `emitErrorCount(0)`, just under the lambda, first had me thinking it was _part_ of the lambda, and I was "wut?".  Just for a second.  :-)  Realizing that is the "seed" for `mergeScan()`, it's not really possible to move it elsewhere, but perhaps the lambda could be extracted to a local function, allowing the `mergeScan()` invocation to be a single-line.

Reply:
Fixed in https://github.com/elastic/kibana/pull/77096/commits/9f6a53bf3aac794d626a9ad20410009a41595af5 by removing usage of arrow function. I think that will do the trick.